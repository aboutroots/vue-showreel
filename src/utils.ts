export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
