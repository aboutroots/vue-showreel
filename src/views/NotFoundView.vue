<template>
  <section class="NotFoundView container">
    <div class="not-found">
      <h1>Sorry, the page you requested was not found!</h1>
      <img src="@/assets/img/404.svg" class="bg-img" alt="image with number 404" />
    </div>
  </section>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'NotFoundView',
});
</script>
<style lang="scss" scoped>
.not-found {
  padding: 2rem;
  text-align: center;
}
</style>
