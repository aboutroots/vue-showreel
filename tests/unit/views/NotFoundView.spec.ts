import NotFoundView from '@/views/NotFoundView.vue';
import { shallowMount } from '../utils';

describe('NotFoundView tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(NotFoundView);

    const container = wrapper.find('.NotFoundView');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(NotFoundView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
