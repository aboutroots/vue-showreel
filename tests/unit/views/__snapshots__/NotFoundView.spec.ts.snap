// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView tests suite matches snapshot 1`] = `
<section
  class="NotFoundView container"
>
  <div
    class="not-found"
  >
    <h1>
      Sorry, the page you requested was not found!
    </h1>
     
    <img
      alt="image with number 404"
      class="bg-img"
      src="@/assets/img/404.svg"
    />
  </div>
</section>
`;
